--- a/net/minecraft/world/entity/animal/Animal.java
+++ b/net/minecraft/world/entity/animal/Animal.java
@@ -295,16 +_,17 @@
         int[][] offsets = DismountHelper.offsetsForDirection(forward);
         BlockPos vehicleBlockPos = this.blockPosition();
         BlockPos.MutableBlockPos targetBlockPos = new BlockPos.MutableBlockPos();
+        final Level level = this.level(); // Gale - Local code optimization - Variable extraction - Animal.getDismountLocationForPassenger() - level
 
         for (Pose dismountPose : passenger.getDismountPoses()) {
             AABB poseCollisionBox = passenger.getLocalBoundsForPose(dismountPose);
 
             for (int[] offsetXZ : offsets) {
                 targetBlockPos.set(vehicleBlockPos.getX() + offsetXZ[0], vehicleBlockPos.getY(), vehicleBlockPos.getZ() + offsetXZ[1]);
-                double blockFloorHeight = this.level().getBlockFloorHeight(targetBlockPos);
+                double blockFloorHeight = level.getBlockFloorHeight(targetBlockPos); // Gale - Local code optimization - Variable extraction - Animal.getDismountLocationForPassenger() - level
                 if (DismountHelper.isBlockFloorValid(blockFloorHeight)) {
                     Vec3 location = Vec3.upFromBottomCenterOf(targetBlockPos, blockFloorHeight);
-                    if (DismountHelper.canDismountTo(this.level(), passenger, poseCollisionBox.move(location))) {
+                    if (DismountHelper.canDismountTo(level, passenger, poseCollisionBox.move(location))) { // Gale - Local code optimization - Variable extraction - Animal.getDismountLocationForPassenger() - level
                         passenger.setPose(dismountPose);
                         return location;
                     }
